### PR TITLE
Move train parser to separate file

### PR DESCRIPTION
**Pull request by schroedtert**
Reintroduce the train file parsing as free functions in the same matter as the event file parsing.

Added simple test for checking if the current functionalities are preserved. Test will be updated and improved with follow up PRs for trains.

### CODE REVIEW

File: libcore/src/Simulation.cpp [hunk 1]
```diff
@@ -556,16 +550,16 @@ double Simulation::RunBody(double maxSimTime)
                 int id           = atoi(strs[1].c_str());
                 std::string type = Trans->GetCaption();
                 trainOutflow[id] += Trans->GetDoorUsage();
-                if(trainOutflow[id] >= TrainTypes[type]->nmax) {
+                if(trainOutflow[id] >= _building->GetTrainTypes().at(type).nmax) {
```
Comment (zeroset) [inline libcore/src/Simulation.cpp:553]: We should catch the exception... Maybe outside this function and Log the Error.

File: libcore/src/Simulation.cpp [hunk 2]
```diff
@@ -1080,22 +1096,22 @@ bool Simulation::TrainTraffic()
...
     auto now    = Pedestrian::GetGlobalTime();
-    for(auto && tab : TrainTimeTables) {
-        trainType  = tab.second->type;
-        trainId    = tab.second->id;
-        trackStart = tab.second->pstart;
-        trackEnd   = tab.second->pend;
-        if(!tab.second->arrival && (now >= tab.second->tin) && (now <= tab.second->tout)) {
-            trainHere                            = true;
-            TrainTimeTables.at(trainId)->arrival = true;
+    for(auto & [id, tab] : _building->GetTrainTimeTables()) {
```
Comment (zeroset) [inline libcore/src/Simulation.cpp:1099]: const?

File: systemtest/train_tests/simple_train_test/runtest_simple_train.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+def runtest(inifile, trajfile):
+    fps, N, traj = parse_file(trajfile)
+
+    maxFrame = np.max(traj[:, 1])
+    time = maxFrame / fps
+
+    if time > 50:
+        logging.critical("%s exists with failure!" % argv[0])
+        logging.critical("simulation took longer than 44s")
```
Comment (zeroset) [inline systemtest/train_tests/simple_train_test/runtest_simple_train.py:27]:
```suggestion
        logging.critical("simulation took longer than 50s")
```

File: libcore/src/Simulation.cpp (hunk 2 above)
Comment (schroedtert) [reply libcore/src/Simulation.cpp]: changed

File: libcore/src/Simulation.cpp (hunk 1 above)
Comment (schroedtert) [reply libcore/src/Simulation.cpp:553]: added try catch around RunStandardSimulation in main.cpp

**Pull request merged**